### PR TITLE
feat: change storing of realm_entity_id to realm_id + add subscription to the SpawnNpc event

### PR DESCRIPTION
**Pull request by edisontim**
Positions and names of the realms are determined by the `realm_id` and not the `realm_entity_id`. Adds the `realm_id` to the smart contracts via the  PR and stores is in this PR.

PR also includes the subscription to the `SpawnNpc` event

Refactored a few things as well 😬

### CODE REVIEW

**Review by cwastche — CHANGES_REQUESTED** [review]
Found a breaking bug in file overlore/llm/natural_language_formatter.py
Function _order_accepted_to_nl()
Line 57 : """metadata: dict[Any, Any] = str_to_json(str(event[6]))"""
Metadata was pushed back to the 8th index.

File: overlore/townhall/service.py [hunk 1]
```diff
@@ -55,41 +64,51 @@ async def service(websocket: WebSocketServerProtocol, config: Config) -> None:
...
 
 
+async def start() -> None:
+    config = setup()
+
+    if config.prompt:
+        await prompt_loop(config)
+
+    # Sync events database on boot
+    await torii_boot_sync(config.env["TORII_GRAPHQL"])
```
Comment (cwastche) [inline overlore/townhall/service.py:107]: torii_boot_sync before launching prompt_loop

**Comment by edisontim** [comment]
> Found a breaking bug in file overlore/llm/natural_language_formatter.py Function _order_accepted_to_nl() Line 57 : """metadata: dict[Any, Any] = str_to_json(str(event[6]))""" Metadata was pushed back to the 8th index.
> 
> Same for _combat_outcome_to_nl() Line 34

Fixed in my latest commit

File: overlore/townhall/service.py [hunk 2]
```diff
@@ -8,32 +8,41 @@
...
+    (Subscriptions.NPC_SPAWNED, assign_name_character_trait_to_npc),
+]
 
 
-async def prompt_loop(config: Config) -> None:
+async def prompt_loop(config: BootConfig) -> None:
     while True:
         txt = input("hit enter to generate townhall with realm_id 73 or enter realm_id\n")
         realm_id = 73 if len(txt) == 0 else int(txt)
         msg = f'{{"realm_id": {realm_id}, "order": 1}}'
```
Comment (cwastche) [inline overlore/townhall/service.py:37]: change "order" to "orderId" as per your latest commit

**Review by cwastche — APPROVED** [review]
Lgtm, small typo in prompt_loop that needs to change

**Review by cwastche — APPROVED** [review]
Lgtm, mergeable

**Pull request merged**